Objects/longobject.c

```diff
@@ -3611,24 +3611,30 @@ long_richcompare(PyObject *self, PyObject *other, int op)
     Py_RETURN_RICHCOMPARE(result, 0, op);
 }
 
+static inline int
+/// Return 1 if the object is one of the immortal small ints
+_long_is_small_int(PyObject *op)
```

Comment:
`_long_is_small_int` could now be used in

https://github.com/python/cpython/blob/67b18a18b66b89e253f38895057ef9f6bae92e7b/Objects/longobject.c#L64-L68

https://github.com/python/cpython/blob/67b18a18b66b89e253f38895057ef9f6bae92e7b/Objects/longobject.c#L204-L210

as well. But maybe this is for a different pull request?

Reply:
Maybe, but how exactly? Both methods above check for a PyLongObject  to have a small value, and then replace it with the built in small int with the same value. The `_long_is_small_int` only checks for the builtin immortal small ints. 

(In other words: two different PyLong objects can have the same value, and we need to check on all)

Reply:
My bad, that indeed won't work there :(